### PR TITLE
Add gtag to carousel

### PR DESCRIPTION
**Pull request by kjgalvan**
Sends promotion data to Google Analytics for the home carousel.

### CODE REVIEW

File: demo/src/screens/HomeScreen/HomeCarousel/HomeCarousel.js [hunk 1]
```diff
@@ -1,16 +1,86 @@
...
+      ],
+      promotion_name: 'Home Carousel',
+    });
+  }, []);
+
+  /**
+   * Send select_promotion event to Google Analytics when the first
+   * carousel slide item is selected.
+   */
+  function onSlideOneClick() {
```
Comment (calhounsm) [inline demo/src/screens/HomeScreen/HomeCarousel/HomeCarousel.js:42]: There's a lot of redundancy between the different onSlideXClick methods here.

WDYT about refactoring to a single `onSlideClicked(itemId, discount = undefined)` and passing arguments  in the onClick handlers?
e.g. https://reactjs.org/docs/handling-events.html#passing-arguments-to-event-handlers

Comment (kjgalvan) [reply demo/src/screens/HomeScreen/HomeCarousel/HomeCarousel.js:42]: This is a good idea, I definitely felt bad about all the copy + pasting! I'll update the PR asap

**Comment by kjgalvan** [comment]
It's ready! Thanks again for the suggestion.

File: demo/src/screens/HomeScreen/HomeCarousel/HomeCarousel.js [hunk 2]
```diff
@@ -1,16 +1,59 @@
...
+   * @param {number=} discount Price discount of item (if any)
+   */
+  function onSlideClick(itemId, discount = 0) {
+    event('select_promotion', {
+      items: [
+        {
+          ...getItemParameters(itemId),
+          promotion_name: 'Home Carousel',
+          // if discount is positive, include it in the item parameter
+          ...(discount > 0 && {discount}),
```
Comment (calhounsm) [inline demo/src/screens/HomeScreen/HomeCarousel/HomeCarousel.js:51]: Now that's some code golf.

**Pull request merged**